lib/queue.rb

```diff
@@ -1,31 +1,49 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = -1
+    @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
-  end
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    if @front == @back
+      #decide
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
 
-  def dequeue
-    raise NotImplementedError, "Not yet implemented"
   end
+  
+  def dequeue
```

Comment:
You also need to check if the queue is empty before and after you dequeue an element.